src/sql/workbench/services/contextualization/common/serverContextualizationService.ts

```diff
@@ -80,14 +81,27 @@ export class ServerContextualizationService extends Disposable implements IServe
 	 * @param ownerUri The URI of the connection to get context for.
 	 */
 	public async getServerContextualization(ownerUri: string): Promise<azdata.contextualization.GetServerContextualizationResult> {
+		const copilotExt = await this._extensionService.getExtension('github.copilot');
+		const isContextualizationEnabled = this._configurationService.getValue<IQueryEditorConfiguration>('queryEditor').githubCopilotContextualizationEnabled;
+
 		const providerName = this._connectionManagementService.getProviderIdFromUri(ownerUri);
 		const handler = this.getProvider(providerName);
-		if (handler) {
+		if (copilotExt && isContextualizationEnabled && handler) {
 			return await handler.getServerContextualization(ownerUri);
 		}
 		else {
 			return Promise.resolve({
-				context: []
+				context: undefined
+			});
+		}
+	}
+
+	public async sendServerContextualizationToCopilot(serverContext: string | undefined): Promise<void> {
+		const copilotExt = await this._extensionService.getExtension('github.copilot');
+		if (copilotExt && serverContext && this._configurationService.getValue<IQueryEditorConfiguration>('queryEditor').githubCopilotContextualizationEnabled) {
+			// LEWISSANCHEZ TODO: Find way to set context on untitled query editor files. Need to save first for Copilot status to say "Has Context"
+			await this._commandService.executeCommand('github.copilot.provideContext', '**/*.sql', {
```

Comment:
1. Is there any way to specify what editor the context is for?
2. Does providing context clear the other context?

Reply:
I can comment here (since I'm working on the copilot fork side): context is associated to files via a glob-style pattern. One context object is allowed per unique pattern (so, in this case, calling `provideContext` again with the exact same `**/*.sql` pattern will _replace_ the existing context). 

You could make this call a bit more granular, so that it matches only the given file (but we still need to work through what happens w/ untitled editors).

Reply:
Cool, thanks for the info. How does copilot know what editor to associate what context with?

Reply:
On the copilot side, as you type in an editor, we'll see if the uri for that editor matches any pattern in the `{ pattern: context }` map. For each pattern that matches, we add the context as a candidate for inclusion into the prompt.

This seemingly works well (except for untitled editors, not sure what the uri for those looks like on the Copilot side, need to do some tests there).

Reply:
Another good question, I'm not doing anything for cleanup and that context map could continue to grow as you open/close unique files (assuming we use file-specific patterns). I doubt this is an issue at this point as the data is tiny, but I could always add a way for ADS to "delete" a context entry when it's no longer needed. 

(Doing the cleanup from my side may be trickier, it'd be easier for whomever registers context to remove it when no longer needed.)

Reply:
Cleanup could be done by setting the context to empty when an editor is closed, right?

Anyways, I agree it's not something that is critical to address now it sounds like - but something to keep in mind as we're designing how this will all fit together.

Reply:
Set to empty (assuming we're using the more specific per-file patterns) would be a way to implement some level of cleanup for now, yes!

Reply:
@Charles-Gagnon, I thought this was already scoped to the file, but guess not. Do you know where I would need to make changes to scope context to a file?

Reply:
@jjhenkel Should be able to help. From what he was saying earlier you have to modify the pattern you use when setting it (right now hardcoded to `**/*.sql`). So the change I believe it would be would be something more like `<Path to file>/FileName.sql` instead (although what the pattern can accept isn't something I know)

Reply:
@lewis-sanchez I'm looking at your branch today/tomorrow and can do some testing w/ stricter path constraints and what happens w/ untitled files (and modify the copilot fork if needed).